src/arp.c

```diff
@@ -70,6 +70,18 @@ int arp_cache_insert(in_addr_t ip_addr,
     if (ip_addr == 0)
         return 0;
 
+    struct arp_cache_entry find = {
+        .ip_addr = ip_addr,
+    };
+
+    if ((entry = RB_FIND(arp_cache_tree, &arp_cache_head, &find)) >0){
+        if (memcmp(entry->haddr, haddr, sizeof(mac_addr_t)) != 0){
```

Comment:
It looks weird. Can you simplify?

Reply:
Yes, I ' ll try to simplify the code. By the way, shall I convert the PR state to draft first?

Reply:
Keep the same pull request for reviewing.